test-runner/src/main/scala/scala/scalanative/testinterface/ComRunner.scala

```diff
@@ -68,24 +70,32 @@ class ComRunner(bin: File,
   /** Wait for a message to arrive from the distant program. */
   def receive(timeout: Duration = Duration.Inf): Message =
     synchronized {
-      in.mark(Int.MaxValue)
       val savedSoTimeout = socket.getSoTimeout()
+
       try {
-        val deadLineMs = if (timeout.isFinite()) timeout.toMillis else 0L
-        socket.setSoTimeout((deadLineMs min Int.MaxValue).toInt)
+        // Java sockets only handle non-negative Int timeouts.
+        // Silently truncate timouts greater than Int.MaxValue.
+        val deadLineMs = if (!timeout.isFinite()) {
```

Comment:
The only caller of `receive()` is at
https://github.com/scala-native/scala-native/blob/b805570124364b58a61d2d43b8e31ed7df639025/test-runner/src/main/scala/scala/scalanative/testinterface/ScalaNativeRunner.scala#L35-L36
which doesn't give a timeout. So instead of improving how the timeout is set, you can just get rid of the `timeout` parameter and all the handling with the timeouts.